svo (install me in module manager).xml

```diff
@@ -7174,36 +7174,29 @@ end</script>
 				<packageName></packageName>
 				<script>function svo.classchange(override)
   local newclass = (override ~= "gmcp.Char.Status") and override or gmcp.Char.Status.class
-  local temp_defs = {}
 	
 	-- if you're in Dragon or an unrecognised class, use the None system
 	-- works fine in Dragon as they don't have class-specific skills anyway
 	if not svo.knownskills[newclass:lower()] then newclass = "None" end
 		
-	-- save general defences in temp table, so svo doesn't waste resources
-	-- deffing what is already up	
-  for k,v in pairs(svo.defc) do
-    if v then
-      if svo.defs_data[k].type then
-        if svo.defs_data[k].type == "general" then
-          temp_defs[k] = true
-        end
-      end
-    end
-  end	
+
 		
   if svo.me.class ~= newclass then
     local oldclass = svo.me.class
 
+    -- kill temporary defense triggers for old class
+    for k,v in pairs(svo.tempDefTriggers) do
+      killTrigger(v)
+    end
+
```

Comment:
Clear the able after killing? Else it'll keep growing

Reply:
That won't clear it as the key is K, how about resetting the table after the loop?